ambry-server/src/integration-test/java/com.github.ambry.server/RouterServerSSLTest.java

```diff
@@ -36,20 +41,41 @@
 public class RouterServerSSLTest {
   private static MockCluster sslCluster;
   private static RouterServerTestFramework testFramework;
+  private static MockClusterMap clusterMap;
 
   @BeforeClass
   public static void initializeTests()
       throws Exception {
     File trustStoreFile = File.createTempFile("truststore", ".jks");
+    String sslEnabledDataCentersStr = "DC1,DC2,DC3";
     Properties serverSSLProps = new Properties();
-    TestSSLUtils.addSSLProperties(serverSSLProps, "DC1,DC2,DC3", SSLFactory.Mode.SERVER, trustStoreFile, "server");
+    TestSSLUtils
+        .addSSLProperties(serverSSLProps, sslEnabledDataCentersStr, SSLFactory.Mode.SERVER, trustStoreFile, "server");
     Properties routerProps = getRouterProperties("DC1");
-    TestSSLUtils.addSSLProperties(routerProps, "DC2,DC3", SSLFactory.Mode.CLIENT, trustStoreFile, "router-client");
+    TestSSLUtils.addSSLProperties(routerProps, sslEnabledDataCentersStr, SSLFactory.Mode.CLIENT, trustStoreFile,
+        "router-client");
     MockNotificationSystem notificationSystem = new MockNotificationSystem(9);
-    sslCluster =
-        new MockCluster(notificationSystem, true, "DC1,DC2,DC3", serverSSLProps, false, SystemTime.getInstance());
+    sslCluster = new MockCluster(notificationSystem, serverSSLProps, false, SystemTime.getInstance());
     sslCluster.startServers();
-    testFramework = new RouterServerTestFramework(routerProps, sslCluster, notificationSystem);
+    clusterMap = sslCluster.getClusterMap();
+    testFramework = new RouterServerTestFramework(routerProps, clusterMap, notificationSystem);
+  }
+
+  @Before
+  public void before() {
+    for (MockDataNodeId m : clusterMap.getDataNodes()) {
+      m.setSslPortRequestedStatus(false);
+    }
+  }
+
+  @After
+  public void after() {
+    for (MockDataNodeId m : clusterMap.getDataNodes()) {
+      if (m.sslPortWasRequested()) {
```

Comment:
This seems like a weak check that might not hold if we fix some things I think.

To elaborate, it seems like the `Router` is the only one using `getPortToConnectTo()` currently. But this need not always stay true. For e.g. [this](https://github.com/linkedin/ambry/blob/master/ambry-replication/src/main/java/com.github.ambry.replication/ReplicationManager.java#L374) looks like duplicate code to me and this function can easily be discarded and  the single call to this function can be replaced by `getPortToConnectTo()`. (@nsivabalan - could you comment on the feasibility of removing the linked function? It looks superfluous to me).

I'm not saying that the change to `ReplicationManager` has to go into this patch (it would be great if it could!), but in the case of this test, since it also brings up servers, making that replacement would make this check invalid. What do you think?